.github/workflows/ci_build.yaml

```diff
@@ -0,0 +1,138 @@
+        key: ${{runner.os}}-QtCache
+
+    - name: Install Qt ${{matrix.qt_version}} with QtWebEngine
+      if: startsWith(matrix.os, 'windows') && startsWith(matrix.build_type, 'qtwebengine') || startsWith(matrix.os, 'macos') && startsWith(matrix.build_type, 'qtwebengine') || startsWith(matrix.os, 'ubuntu') && !startsWith(matrix.qt_version, 'repository') && startsWith(matrix.build_type, 'qtwebengine')
+      uses: jurplel/install-qt-action@v2
+      with:
+        version: ${{matrix.qt_version}}
+        modules: qtwebengine
+        cached: ${{steps.cache-qt.outputs.cache-hit}}
+
+    - name: Install Qt ${{matrix.qt_version}} without QtWebEngine
+      if: startsWith(matrix.os, 'windows') && startsWith(matrix.build_type, 'qtwebkit') || startsWith(matrix.os, 'macos') && startsWith(matrix.build_type, 'qtwebkit') || startsWith(matrix.os, 'ubuntu') && !startsWith(matrix.qt_version, 'repository') && startsWith(matrix.build_type, 'qtwebkit')
+      uses: jurplel/install-qt-action@v2
+      with:
+        version: ${{matrix.qt_version}}
+        cached: ${{steps.cache-qt.outputs.cache-hit}}
+
+    - name: Install hunspell dependencies on Windows
+      if: startsWith(matrix.os, 'windows') && endsWith(matrix.build_type, 'with hunspell')
+      uses: msys2/setup-msys2@v2
+      with:
+        install: >-
+          base-devel
+          mingw-w64-x86_64-toolchain
+          mingw-w64-x86_64-libtool
+
+    - name: Install hunspell on Windows
+      if: startsWith(matrix.os, 'windows') && endsWith(matrix.build_type, 'with hunspell')
+      shell: msys2 {0}
+      run: |
+        curl -Ls https://github.com/hunspell/hunspell/files/2573619/hunspell-1.7.0.tar.gz -o hunspell.tar.gz
+        tar -xf hunspell.tar.gz
+        cd hunspell-1.7.0
+        autoreconf -vfi
+        ./configure
+        make -j2
+        echo "CMAKE_BUILD_EXTRA=-DHunspell_INCLUDE_DIR=$RUNNER_WORKSPACE\hunspell-1.7.0\src\ -DHunspell_LIBRARIES=hunspell-1.7" >> $GITHUB_ENV
+
+    - name: Install dependencies
+      shell: bash
+      run: |
+
+        # Install APT dependencies on Ubuntu
+        if [[ "${{ matrix.os }}" = "ubuntu"* ]]; then
+          echo "Updating apt repository index"
+          sudo apt update
+          echo "Installing apt packages"
+          sudo apt install -y ${{ matrix.apt-dependencies }}
+        fi
+
+        # Install Hunspell on MacOS
+        if [[ "${{ matrix.os }}" = "macOS"* && "${{ matrix.build_type }}" != *"without"* ]]; then
+          echo "Installing Hunspell"
+          brew install hunspell
+        fi
+
+    - name: Create Build Environment
+      shell: bash
+      run: cmake -E make_directory "${{runner.workspace}}/build"
+
+    - name: Configure CMake
+      working-directory: ${{runner.workspace}}/build
+      shell: bash
+      run: cmake $GITHUB_WORKSPACE $CMAKE_BUILD_EXTRA
+
+    - name: Build Otter Browser
+      working-directory: ${{runner.workspace}}/build
+      shell: bash
+      run: cmake --build . --parallel 3
```

Comment:
I assume you based this on something, but I also assume GHA has a proper variable for it instead?

Reply:
I assume you mean the thread count? Yes: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
I don't think there is a GHA specific variable for the thread count, but my reasoning for that assumption is that you could just get the thread count from the system instead. We have 3 threads hardcoded on the Vircadia GHA which I adopted here.

Reply:
It just doesn't seem very forward thinking. I'd rather see for example $(nproc) or something along those lines. (Presumably this is nproc + 1; is that better somehow?)

Reply:
Now that you talk about $(nproc) I understand why you were talking about a GHA variable and why we have it hardcoded on the Vircadia actions.  I forgot that $(nproc) will only work on Linux (and *maybe* macOS).
GHA doesn't have a variable for the available threads.